operator/pkg/types/declaritive.go

```diff
@@ -24,23 +23,6 @@ type BaseCustomObject interface {
 // ObjectTransform is an operation that transforms the manifest objects before applying it.
 type ObjectTransform = func(context.Context, BaseCustomObject, *ManifestResources) error
```

Comment:
If I understand correctly, BaseCustomObject is equal to `runtime.Object`. Why choose to separate?

Reply:
+`metav1.Object`
```
type BaseCustomObject interface {
	runtime.Object
	metav1.Object
}
```
The idea is to have a generic object both for `typed` and `unstructured` objects